### DISCUSSION TITLE
Handling missing ford/culvert/bridge

### DISCUSSION THREAD

**Discussion by matkoniecz** [streetcomplete/StreetComplete#2426] [Ideas]
One of things often missing and requiring local survey is crossing of waterway and road/path without neither bridge nor culvert nor ford.

It would be a bit tricky to implement as 

- it would be necessary to have quest directly at crossing of waterway and road/path while holding references to both
- in the first step user would need to select what is at a given point: bridge? culvert? ford?
  - for ford: add node both within road and and waterway, tag it properly
  - for culvert: split waterway to mark where culvert starts and ends, tag this section
  - for bridge: split roads to mark where bridge starts and ends, tag this section

I admit that I have no idea where to even start implementing this.

**Comment by westnordost** [comment]
Yeah, this would be very much effort, cause:
- a quest only references one element, the architecture would need to be changed here
- the actions what needs to be done is somewhat diverse, as you describe. Splitting ways, marking the correct position where it starts, adding nodes, etc. This is so complex, I don't even know how to start. Also, how should the user mark the end and start of a bridge/culvert if he doesn't have a satellite image?

What might be possible is to **always** leave an (automatically generated) note in this case. But even then, there is a problem: A quest always references one element and the pin is placed on the center on it. Now, that center will oftentimes not be at the point of the intersection which will confuse users. To solve this, probably it would be necessary to define a completely new quest group (quest groups currently: osm element, osm note) with an own table in the database, maybe an own uploader class etc.

**Reply by westnordost** [reply]
Maybe this is an idea for a new class of quests: "Re: Osmose" quests - Reply to those validation errors that must be surveyed locally.

**Reply by matkoniecz** [reply]
I thought also about this, as someone processing notes I would also appreciate photo of situation from the ground in such cases.

It would turn frustrating QA report into something armchair mappable.

**Reply by westnordost** [reply]
Well maybe a photo would be obligatory for that class of quest.

**Comment by matkoniecz** [comment]
See also https://github.com/streetcomplete/StreetComplete/issues/3749

**Comment by mcliquid** [comment]
Wouldn't it be possible to imitate the functionality of the iD Editor here?

When a path meets a waterway it asks in iD:
- Add a bridge
- Add a tunnel
- Add a Ford

The bridge is simply placed over the river with an approximate length of 10 meters. This will automatically split the highway at a certain length and set the attributes automatically.
The tunnel is set according to the same scheme. The stream is cut up at a length of about 10 meters and the attributes are set accordingly on the new piece.
For the Ford, a point is added exactly on the intersection and this is declared as ford.

Would it be possible to take the basic logic from the iD Editor here?

I suspect at least the issue identification happens here: https://github.com/openstreetmap/iD/blob/021107cb810704df009c0a0a1ad1a28f9349b9d5/test/spec/validations/crossing_ways.js

**Reply by matkoniecz** [reply]
It would require massive effort to implement this.
